pydantic/fields.py

```diff
@@ -310,34 +391,50 @@ class MyModel(pydantic.BaseModel):
         Returns:
             An instance of the field metadata.
         """
-        final = False
-        if _typing_extra.is_finalvar(annotation):
-            final = True
-            if annotation is not typing_extensions.Final:
+        # 1. Check if the annotation is the `Final` type qualifier:
+        final = _typing_extra.is_finalvar(annotation)
+        if final:
+            if _typing_extra.is_generic_alias(annotation):
+                # The annotation is a parametrized `Final`, e.g. `Final[int]`.
+                # In this case, `annotation` will be `int`:
                 annotation = typing_extensions.get_args(annotation)[0]
-
-        if _typing_extra.is_annotated(annotation):
-            first_arg, *extra_args = typing_extensions.get_args(annotation)
-            if _typing_extra.is_finalvar(first_arg):
-                final = True
-            field_info_annotations = [a for a in extra_args if isinstance(a, FieldInfo)]
-            field_info = FieldInfo.merge_field_infos(*field_info_annotations, annotation=first_arg)
+            else:
+                # The annotation is a bare `Final`. Use `Any` as a type annotation:
+                return FieldInfo(annotation=Any, frozen=True)  # pyright: ignore[reportArgumentType] (PEP 747)
+
+        # 2. Check if the annotation is an `Annotated` form.
+        #    In this case, `annotation` will be the annotated type:
+        annotation, metadata = _unpack_annotated(annotation)
```

Comment:
This replaces:

```python
if _typing_extra.is_annotated(annotation):
    first_arg, *extra_args = typing_extensions.get_args(annotation)
```

if `metadata == []`, it means `_typing_extra.is_annotated(annotation) == False`.